nodebuilder/store.go

```diff
@@ -144,29 +138,31 @@ func (f *fsStore) Datastore() (_ datastore.Batching, err error) {
 	// TODO(@Wondertan): Make configurable with more conservative defaults for Light Node
 	opts.MaxTableSize = 64 << 20
 
-	f.data, err = dsbadger.NewDatastore(dataPath(f.path), &opts)
+	ds, err := dsbadger.NewDatastore(dataPath(f.path), &opts)
 	if err != nil {
 		return nil, fmt.Errorf("node: can't open Badger Datastore: %w", err)
 	}
 
-	return f.data, nil
```

Comment:
f.data wasn't nil because it contains the type, and thus the nil check on Close didn't work.